nixos/modules/services/misc/pufferpanel.nix

```diff
@@ -0,0 +1,401 @@
+          {option}`extraPackages` with this option disabled.
+
+          [templates]: https://github.com/PufferPanel/templates
+        '';
+      };
+
+      dynamicLinker = lib.mkOption {
+        type = lib.types.nullOr lib.types.package;
+        default = stdenvDynamicLinker;
+        defaultText = lib.literalMD ''
+          Package containing {file}`lib` directory with symlink to dynamic linker
+          specified in `''${pkgs.stdenv.cc}/nix-support/dynamic-linker`.
+        '';
+        description = lib.mdDoc ''
+          Dynamic linker made available at `/lib` (and `/lib64` on 64-bit
+          systems) for running binaries downloaded by PufferPanel. These
+          directories are private to the service’s filesystem namespace.
+        '';
+      };
+    };
+
+    environment = lib.mkOption {
+      type = lib.types.attrsOf lib.types.str;
+      default = { };
+      example = lib.literalExpression ''
+        {
+          PUFFER_DAEMON_CONSOLE_BUFFER = "1000";
+          PUFFER_DAEMON_CONSOLE_FORWARD = "true";
+          PUFFER_TOKEN_PUBLIC = "https://panel.example.com/auth/publickey";
+        }
+      '';
+      description = lib.mdDoc ''
+        Environment variables to set for the service. Secrets should be
+        specified using {option}`environmentFile`.
+
+        Refer to the [PufferPanel source code][] for the list of available
+        configuration options. Variable name is an upper-cased configuration
+        entry name with underscores instead of dots, prefixed with `PUFFER_`.
+        For example, `panel.settings.companyName` entry can be set using
+        {env}`PUFFER_PANEL_SETTINGS_COMPANYNAME`.
+
+        [PufferPanel source code]: https://github.com/PufferPanel/PufferPanel/blob/master/config/entries.go
+      '';
+    };
+
+    environmentFiles = lib.mkOption {
+      type = lib.types.listOf lib.types.path;
+      default = [ ];
+      description = lib.mdDoc ''
+        Files to load environment variables from. Loaded variables override
+        values set in {option}`environment`.
+      '';
+    };
+
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Open ports in the firewall for PufferPanel.
+      '';
+    };
+
+    host = lib.mkOption {
+      type = lib.types.str;
+      default = "";
+      example = "::1";
+      description = lib.mdDoc ''
+        Bind to the specified IPv4 or IPv6 address.
+      '';
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 8080;
+      description = lib.mdDoc ''
+        Server listen port.
+      '';
+    };
+
+    panel = {
+      enable = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = lib.mdDoc ''
+          Whether to enable PufferPanel web interface.
+        '';
+      };
+
+      registrationEnabled = lib.mkOption {
+        type = lib.types.nullOr lib.types.bool;
+        default = null;
```

Comment:
What means null here? That the overwrite is disabled?

Reply:
Improved docs for this option.

```
This option overrides corresponding setting set via web interface
after service restart. If set to `null` (default), PufferPanel default
configuration is used (registration is enabled but can be disabled in
settings). If set to a boolean value, registration can be temporarily
toggled (until restart) in settings.
```